deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Builds.java

```diff
@@ -210,11 +210,17 @@ public Response publish(
                 "Artifact URL points to unapproved location.");
         }
 
+        // Check if build SCM is approved via allow list of SCMs
+        if (!sourceControlManagerProxy.hasSCMType(buildBean.getScm())) {
```

Comment:
It looks to me that the if the configured type name is "Github" and "github" is received, then the service throws an error. Is it the intention?

Reply:
Correct, that is the intention.

Reply:
I think we should actually allow this by attempt to convert user input to the same case. The conversion is very trivial. There are 2 types of invalid SCM types - invalid case and unsupported type. We do want to avoid unsupported type but should do our best to handle invalid case.

Reply:
Okay then that change would need to be made to the BuildBean class that accepts the scm json value. I would have to reassign the scm initialized here:
`@JsonProperty("type")
    private String scm;`
to a capitalized version.

Reply:
Yes updating the setter of `private String scm` to capitalize all setter input is a good point. That ensures only capitalized SCM types are stored in the DB from now on. We should also update the config so `typeName` of each manager is capitalized. Last we want to capitalize user input. This may be an incomplete list, please ensure all other necessary scenarios are updated accordingly.

Reply:
Sounds good. I already have a PR in Phabricator for the config updates you mentioned. I will implement what you mentioned here.